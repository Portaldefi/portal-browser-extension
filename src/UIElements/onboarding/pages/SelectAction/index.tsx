import React, { useEffect, useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Button, Card, Grid, GridColumn, GridRow, Header, Icon } from 'semantic-ui-react';

import { setItems } from '../../slices/menuSlice';
import { useAppDispatch } from '../../hooks';

export default () => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(setItems([
      {
        title: 'Back',
        link: '/welcome',
      }
    ]));
  }, []);

  const handleImportWallet = useCallback(() => {
    navigate('/import-wallet');
  }, []);

  const handleCreateWallet = useCallback(() => {
    navigate('/create-wallet')
  }, []);

  return (
    <Grid>
      <GridRow>
        <GridColumn width='sixteen'>
          <Header className='' as='h1' inverted>New to Fabric Wallet?</Header>
        </GridColumn>
      </GridRow>
      <GridRow>
        <GridColumn width='eight'>
          <Card fluid>
            <Card.Content>
              <Card.Header>
                <Icon name='download' size='big' />
                <Header as='h2'>No I already have a Secret Recovery Phrase</Header>
              </Card.Header>
              <Card.Description>
                <div>Import your existing wallet using a Secret Recovery Phrase</div>
              </Card.Description>
            </Card.Content>
            <Card.Content extra>
              <Button color='blue' size='massive' onClick={handleImportWallet}>
                Import Wallet
              </Button>
            </Card.Content>
          </Card>
        </GridColumn>
        <GridColumn width='eight'>
          <Card fluid>
            <Card.Content>
              <Card.Header>
                <Icon name='plus' size='big' />
                <Header as='h2'>Yes, letâ€™s get set up!</Header>
              </Card.Header>
              <Card.Description>
                <div>This will create a new wallet and Secret Recovery Phrase</div>
              </Card.Description>
            </Card.Content>
            <Card.Content extra>
              <Button color='blue' size='massive' onClick={handleCreateWallet}>
                Create a Wallet
              </Button>
            </Card.Content>
          </Card>
        </GridColumn>
      </GridRow>
    </Grid>
  )
}