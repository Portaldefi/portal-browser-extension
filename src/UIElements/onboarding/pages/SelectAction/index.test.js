import SelectAction from './index';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

describe('SelectActionComponent', () => {
  it('should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
            <SelectAction />
        </BrowserRouter>,
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});