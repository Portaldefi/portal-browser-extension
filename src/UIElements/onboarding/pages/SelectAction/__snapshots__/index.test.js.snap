// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectActionComponent should work 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        Connect Identity
      </div>
    </div>
    <div
      className="centered row"
    >
      <p
        className="ui grey small header description"
      >
        Every identity is created with a seed, with the option to create multiple detached identities from each seed.
      </p>
    </div>
    <div
      className="centered row"
    >
      <img
        className="ui image"
        src="/images/onboarding/vault.png"
      />
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <button
          className="ui button secondary-button"
          onClick={[Function]}
        >
          Import Seed
        </button>
      </div>
      <div
        className="column"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Create New Identity Seed
        </button>
      </div>
    </div>
  </div>
</div>
`;
